### ISSUE TITLE
Error with class inheriting from generic interface twice

### ISSUE THREAD

**Issue by starkcolin** [autofac/Autofac#870]
Version 4.6.1

When a class inherits from the same generic interface twice, an error message like below is shown:
    The implementation type CGeneric'1 does not support the interface ISingleGenericWithInModifier'1.

The following test should pass:

```csharp
public class TestClass
{
    public interface ISimpleInterface
    {
    }

    public interface ISingleGeneric<T>
    {
    }

    public interface ISingleGenericWithInModifier<in T>
    {
    }

    public interface ISingleGenericWithOutModifier<out T>
    {
    }

    public class CGeneric<T> : ISingleGeneric<T>
    {
    }

    public class CSimple : ISimpleInterface
    {
    }

    public class CDoubleInheritGeneric<T> : ISingleGenericWithInModifier<ISingleGeneric<T>>, ISingleGenericWithInModifier<CGeneric<T>>
    {
    }

    [Fact]
    public void CanResolveComponentWhenItInheritsFromTheSameGenericInterfaceTwice()
    {
        var builder = new ContainerBuilder();

        builder.RegisterGeneric(typeof(CDoubleInheritGeneric<>)).As(typeof(ISingleGenericWithInModifier<>));

        var container = builder.Build();

        // These should resolve because it inherits from ISingleGenericWithInModifier with both as nested generic arguments
        Assert.True(container.IsRegistered<ISingleGenericWithInModifier<ISingleGeneric<int>>>());
        Assert.True(container.IsRegistered<ISingleGenericWithInModifier<CGeneric<int>>>());
        Assert.True(container.IsRegistered<ISingleGenericWithInModifier<ISingleGeneric<CSimple>>>());
        Assert.True(container.IsRegistered<ISingleGenericWithInModifier<CGeneric<CSimple>>>());

        // These should not resolve because they do not match the above pattern
        Assert.False(container.IsRegistered<ISingleGenericWithInModifier<int>>());
        Assert.False(container.IsRegistered<ISingleGenericWithInModifier<ISingleGenericWithOutModifier<int>>>());
        Assert.False(container.IsRegistered<ISingleGenericWithInModifier<ISingleGenericWithOutModifier<CSimple>>>());
    }
}
```

The fix is to change the call from Single to First on line 109 of OpenGenericServiceBinder in method GetInterface.
I have tested it locally and all tests pass.

I have created [PR #871 ](https://github.com/autofac/Autofac/pull/871) with the fix and test for the issue

**Comment by alexmg** [comment]
Fixed by commit e58299b2d440cc9e8de23c67ebefa2574dd05312.

**Comment by tillig** [comment]
Released as 4.6.2.
